pandas/core/internals/managers.py

```diff
@@ -366,6 +367,37 @@ def reduce(
             new_mgr = type(self).from_blocks(res_blocks, [self.items, index])
         return new_mgr, indexer
 
+    def operate_scalar(self, other, op) -> BlockManager:
+        """
+        TODO fill in
+        """
+        # Get the appropriate array-op to apply to each column/block's values.
+        array_op = ops.get_array_op(op)
+        return self.apply(array_op, right=other)
+
+    def operate_array(self, other: ArrayLike, op, axis: int) -> BlockManager:
```

Comment:
Note I currently add 3 specialized methods (`operate_scalar`, `operate_array` and `operate_manager`) for the three possible cases that are left to handle here. 
But it could of course also be a single `operate` method that combines the three with some if/elif/else checks (but those checks would duplicate a bit the checks done in `_dispatch_frame_op` where those get called)